packages/components/src/components/toast/toast.tsx

```diff
@@ -3,7 +3,24 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import { CSSTransition } from 'react-transition-group';
 
-const Toast = ({ children, className, is_open = true, onClose, onClick, type = 'info', timeout = 0 }) => {
+type TToast = {
+    className: string;
+    is_open: boolean;
+    onClick: (event: React.MouseEvent<HTMLElement>) => void;
+    onClose: () => void;
+    type: 'error' | 'info';
+    timeout: number;
+};
+
+const Toast = ({
+    children,
+    className,
+    is_open = true,
+    onClose,
+    onClick,
+    type = 'info',
+    timeout = 0,
+}: React.PropsWithChildren<Partial<TToast>>) => {
```

Comment:
@likhith-deriv Why use `Partial` here? 🤔 If all the properties are optional then why not make them optional in the `TToast` type for better readability? 🤔

Reply:
I thought of using the interfaces that React provides. I don't think Partial makes the code less readable

Reply:
@likhith-deriv `Partial` is a utility type from TypeScript, Not React. We mainly use it in test files for mock data where you have a big object and you only need part of it for the test.

The props type of a component defines the signature of that component, As a dev, I will be looking at the type definition to understand what this component needs to work, So if I check the type I will see everything as required and I can easily miss the part `Partial` has been used, So by doing this we are making devs to look for two places to understand how things work which we are losing the code reality.

It's nothing important tho 😅 Let's move forward 🙇🏻